cvss3/score.go

```diff
@@ -23,12 +24,33 @@ func roundUp(x float64) float64 {
 	return math.Ceil(x*10) / 10
 }
 
+// Validate should be called before calculating any scores on vector
+// If there's an error, there's no guarantee that a call to *Score() won't panic
+func (v Vector) Validate() error {
+	for metricName, def := range map[string]defineable{
```

Comment:
Did you forget to actually call it? I cannot find any references to this function anywhere in the code.
Also, creating this map and ranging over it is *way* slower than a `switch` statement, so if it is to be called for every `Score()` call, it's better to use switch. Try benchmarking it, you'd be surprised (I was).

Reply:
I wasn't going for speed, but ok. These seem like micro optimizations and degrade the readability and maintainability. I'll change it to a switch

Reply:
If I got correctly that it is going to be called at least once per each `Score()` call, then, considering the target usecase is stream processing, the performance boost could be significant. Similar thing shaved off ~10% of `cpe2cve` runtime at some point. And again, that's just an observation, no premature optimisation is required, obviously.

Reply:
You got it correctly, of course :) it's 6x faster (90ns/op -> 560ns/op on my machine). 
But i thought we were developing enterprise software here, m8.

Reply:
This was a *very* inappropriate thought, m8 :)